plugins/rapture/nexus-coreui-plugin/src/test/java/org/sonatype/nexus/coreui/RepositoryXOTest.groovy

```diff
@@ -0,0 +1,103 @@
+package org.sonatype.nexus.coreui
+
+import javax.validation.Validation
+import javax.validation.Validator
+
+import org.sonatype.nexus.validation.group.Create
+
+import spock.lang.Shared
+import spock.lang.Specification
+import spock.lang.Unroll
+
+class RepositoryXOTest
+    extends Specification
+{
+
+  @Shared
+  Validator validator = Validation.buildDefaultValidatorFactory().validator
+
+  static List INVALID_CHARACTERS = ['_', '#', ',', 'A', '*', ' ', '#', '\'', '\\', '/', '?', '<', '>', '|', ' ', '\r',
+                                    '\n', '\t', ',', '.'].asImmutable()
```

Comment:
If we expect anyone ever to use non-english characters, then no; although that presents its own problems. I opted to just negate characters that are 'known bad' either to elasticsearch for index naming, or for URLs(incomplete right now as @mcculls mentioned)